### PR TITLE
preparing sve top bits

### PR DESCRIPTION
**Pull request by DenisYaroshevskiy**
* ripped out avx512 top bits.
* redid top bits for avx512 as a logical inside.
* added a few sve operators
* fixed the testing for logicals

### CODE REVIEW

File: include/eve/detail/function/simd/arm/sve/make.hpp [hunk 1]
```diff
@@ -98,7 +98,7 @@ make(eve::as<logical<wide<T, N>>>, V x) noexcept requires sve_abi<abi_t<T, N>> &
   {
     return [&]<std::size_t... i>(std::index_sequence<i...> const&)
     {
-      return make(as<logical<wide<T, f_t>>> {}, (i < N::value ? x : false)...);
+      return make(as<logical<wide<T, f_t>>> {}, (i < N::value ? (bool)x : false)...);
```
Comment (jfalcou) [inline include/eve/detail/function/simd/arm/sve/make.hpp:101]: I would prefer `static_cast`.

File: include/eve/arch/cpu/top_bits.hpp [hunk 2]
```diff
@@ -0,0 +1,488 @@
...
+  } // namespace eve
+
+// ---------------------------------------------------------------------------------
+// to_logical(top_bits)
+//
+
+namespace eve::detail {
+
+template <logical_simd_value Logical>
+EVE_FORCEINLINE Logical to_logical(eve::top_bits<Logical> mmask) noexcept
```
[the remark this answers is not in the document]
Comment (DenisYaroshevskiy) [reply include/eve/arch/cpu/top_bits.hpp:412]: the problem is - predeclaration of to_logical
We need then to predeclare to separate versions: just for top_bits and for top_bits on x86

We should always use callables. Or do some other trick. For now let's leave this.

File: include/eve/detail/function/simd/arm/sve/make.hpp (hunk 1 above)
Comment (DenisYaroshevskiy) [reply include/eve/detail/function/simd/arm/sve/make.hpp:101]: If you ok with the other thing, let's do in the next patch

File: include/eve/arch/cpu/top_bits.hpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (jfalcou) [reply include/eve/arch/cpu/top_bits.hpp:412]: OK I see

File: include/eve/detail/function/simd/arm/sve/make.hpp (hunk 1 above)
Comment (jfalcou) [reply include/eve/detail/function/simd/arm/sve/make.hpp:101]: Fine

**Review by jfalcou — APPROVED** [review]

**Pull request merged**